include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -736,6 +736,19 @@ inline Value Value::CreateSequence(std::vector<Value>& values) {
   return Value{out};
 }
 
+inline Value Value::CreateEmptyTensorSequence(ONNXTensorElementDataType tensor_type) {
+  OrtValue* out;
+  ThrowOnError(GetApi().CreateEmptyTensorSequence(tensor_type, &out));
+  return Value{out};
+}
+
+template <typename T>
+inline Value Value::CreateEmptyTensorSequence() {
+  OrtValue* out;
+  ThrowOnError(GetApi().CreateEmptyTensorSequence(TypeToTensorType<T>::type, &out));
+  return Value{out};
+}
+
```

Comment:
What's the use case where someone using the API needs to create an empty sequence?

Reply:
Apart from the node test case for opst-13 loop which consumes an empty sequence as a graph input (and populates the sequence every iteration), it does seem weird that currently the APIs only allow creation of sequences that have atleast one element (seems like a very strict constraint to arbitrarily impose on any user)

Reply:
Not quite following. When does a user need to use the API to create an empty sequence? Seems like the model could provide a default empty sequence in a non-const initializer if that was needed.

8KB is a significant binary size increase for these changes, so we should trim everything we can. If we need a way to create an empty sequence (not convinced that we do yet), at the very least the templatized API method is not required.

Reply:
When does a user need to use the API to create an empty sequence?
Whenever the user has a model that consumes an empty sequence. The test model was just a specific illustration. It scales to any model that consumes an empty sequence. Yes, the model could use a non-const initializer but that is kind of beside the point for me. It fixes this specific issue but doesn't fix the gap in the API. The API allows creation of sequences in general but only as long as they have atleast one element. Don't you think that is an arbitrary constraint given that ONNX itself allows creating empty sequences ?

Reply:
I haven't seen a single real world model so far that uses tensor sequences, let alone models that take empty tensor sequences as inputs. Hence this wasn't supported before. Given the existence of CreateValue API that supports non-tensor types (non-empty), I would say let's think this more before committing it as a public API. The worst is you will not be able to run one artificial node test.

Reply:
This is the distribution of the 7~8 KB that this change adds:

![image](https://user-images.githubusercontent.com/9969784/98423286-0c86f180-2043-11eb-8558-16febca431b0.png)

As you can see the bulk of the size (> 5KB) comes from the op changes and other miscellaneous pieces needed. The API adds <  2KB and the value addition it provides is that it adds some completeness to sequence creation using the ORT API. It just feels weird that we allow creating sequences in general but we draw the line just before creating empty sequences. Having said that, we could pull out the API and save 2KB and move the node test to the filtered list as well and that is fine too.

Reply:
Pulled out the API + made changes in the test runner accordingly